### PR TITLE
Cluster provision refactor

### PR DESCRIPTION
**Pull request by chetnieter**
This refactor moves much of the redundant code in the various taskflows to a new intermediate class, ClusterProvisioningTaskFlow, in cumulus. This PR depends on Kitware/cumulus#268.

### CODE REVIEW

File: server/taskflows/hpccloud/taskflow/paraview.py [hunk 1]
```diff
@@ -87,26 +65,14 @@ def start(self, *args, **kwargs):
         kwargs['image_spec'] = image_spec
```
Comment (chetnieter) [inline server/taskflows/hpccloud/taskflow/paraview.py]: This seems redundant with the line above. I assume this statement should only be called once but should it be in the if block or not?

Comment (cjh1) [inline server/taskflows/hpccloud/taskflow/paraview.py]: I agree 85 should be removed.

File: server/taskflows/hpccloud/taskflow/pyfr.py [hunk 2]
```diff
@@ -111,25 +89,7 @@ def start(self, *args, **kwargs):
```
Comment (chetnieter) [inline server/taskflows/hpccloud/taskflow/pyfr.py]: @cjh1 Should line 88 be moved out of the if block?

Comment (cjh1) [inline server/taskflows/hpccloud/taskflow/pyfr.py]: Sorry I'm not clear what you mean?

Comment (chetnieter) [inline server/taskflows/hpccloud/taskflow/pyfr.py]: The assignment "kwargs['image_spec'] = image_spec" appears in two [places](https://github.com/Kitware/HPCCloud/blob/cluster-provision-refactor/server/taskflows/hpccloud/taskflow/paraview.py#L63-L65) in paraview.py. In the first location it is contained in the "if '_id' not in kwargs['cluster']:" if block. Based on your response above I assume to should not be in that if block.

There is only one "kwargs['image_spec'] = image_spec" [assignment](https://github.com/Kitware/HPCCloud/blob/cluster-provision-refactor/server/taskflows/hpccloud/taskflow/pyfr.py#L88) in pyfr.py but it is also inside a similar if block. Should it be in that if block or should that assignment always happen?

Comment (cjh1) [inline server/taskflows/hpccloud/taskflow/pyfr.py]: The assignment should always happen.

**Comment by chetnieter** [comment]
@cjh1 PTAL - I was able run and visualize a pyfr simulation with this branch.

**Comment by cjh1** [comment]
LGTM

**Pull request merged**